### PR TITLE
Integrated Flask-Babel 

### PR DESCRIPTION
**Pull request by esegal**
Added Babel, added first 3 strings to test translation, added [lang] GET parameter to change language.

It doesn't complete the translate task but it allows everyone to start using Babel for multilanguage in anyway. 

* First, install Flask-Babel (it's added to requirements.txt, so you can `pip -r requirements.txt`

In the python code, instead of writing a message string like `'Message'`, use `gettext(u'Message In English')` (see `main.py`)

In the templates, instead of writing strings like `<title>ANYWAY</title>`, use `<title>{{ gettext('ANYWAY - Influencing in Any Way') }}</title>` (See `templates/index.html`)

After adding strings they will appear in English as they are (Babel's instructions suggest these strings be English). if you wish to add Hebrew translation you should extract the new strings you added by using this command:

```$ pybabel extract -F babel.cfg -o messages.pot .```

Then, you need to merge this strings into the Hebrew translation file:

```$ pybabel update -i messages.pot -d translations```

Then, edit translationa/he/LC_MESSAGES/messages.po, and translate the strings there. Finally, to compile your changes run

```$ pybabel compile -d translations```

See https://pythonhosted.org/Flask-Babel/ for more.

### CODE REVIEW

File: messages.pot [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 1.3\n"
+
+#: main.py:181
+msgid "Discussion not found:"
+msgstr ""
```
Comment (danielhers) [inline messages.pot]: Please add the translations here for the Hebrew messages you replaced (just copy from the original)

File: templates/index.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>ANYWAY - משפיעים בכל דרך</title>
+    <title>{{ gettext('ANYWAY - Influencing in Any Way') }}</title>
```
Comment (danielhers) [inline templates/index.html]: @galraij is this a good translation for the slogan? :)

File: main.py [hunk 3]
```diff
@@ -175,9 +188,9 @@ def discussion():
                             identifier).first()
                 context['title'] = marker.title
             except AttributeError:
-                return index(message=u"הדיון לא נמצא: " + request.values['identifier'])
+                return index(message=gettext(u'Discussion not found:') + request.values['identifier'])
```
Comment (danielhers) [inline main.py]: @yosefh Doesn't work: `NameError: global name 'gettext' is not defined`

**Comment by danielhers** [comment]
@esegal looks good, but please see comments and fix before I merge.
Also, perhaps it's a good idea to add the compiled messages file to version control (I think that's the standard, even though it's automatically generated). Otherwise the app doesn't show the translated messages out of the box, as you can see in the [review app](https://anyway-unstable-pr-438.herokuapp.com/): it shows the English messages.

**Comment by esegal** [comment]
I fixed all the issues

File: templates/index.html [hunk 4]
```diff
@@ -1,15 +1,14 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>ANYWAY - משפיעים בכל דרך</title>
+    <title>{{ gettext('ANYWAY - Making a difference in any way') }}</title>
```
Comment (danielhers) [inline templates/index.html]: @esegal I'm still getting this message in English when I load the site. What could be the problem?

**Pull request merged**